src/honeysql_postgres/format.cljc

```diff
@@ -113,6 +114,13 @@
   (str "DO UPDATE SET " (sqlf/comma-join (for [[k v] values]
                                            (str (sqlf/to-sql k) " = " (sqlf/to-sql v))))))
 
+(defmethod format-clause :filter [[_ expr] sql-map]
```

Comment:
Couple of things to note:

1. Nested threaded macros are generally hard to follow, and especially so when the arguments are flipping positions. I try to avoid macros entirely within a threaded expression. 
2. The `as->` expression here will have to be bound to something (I think that’s been missed), and that binding will go unused. In the first `as->` expression, the  alias doesn’t undergo further transformation. And in the second `as->` expression, the binding would be redundant since it’s not used.
3. With string concatenation/formatting that require certain strings to be concatenated/formatted in conditionally, it’s generally easier to follow when the arguments are in-place, as opposed to when it’s done using threaded macros. 

Let me know if I’ve missed the rationale of doing it this way or if you have other thoughts about the things I’ve noted here.

Reply:
@scimetfoo I agree for complex threads, but I felt like this was a simple enough case to fit in well. Although my usage was clearly broken, so what do you think of threading all the other expressions except for the conditional `as->`?

Reply:
@kitallis what are your thoughts on this?

```
(defmethod format-clause :filter [[_ expr] sql-map]
  (let [format (fn [expr]
                 (let [[expression clause alias] (mapv sqlf/to-sql expr)]
                   (->> alias
                        (str " AS ")
                        (when alias)
                        (str expression " FILTER " clause))))]
    (->> expr
         (map format)
         sqlf/comma-join
         (str (when (seq (:select sql-map)) ", ")))))
```

Reply:
@scimetfoo Yup looks good to me 👍🏼